extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -319,9 +342,68 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
   }
 
   /** @override */
+  buildCallback() {
+    super.buildCallback();
+
+    this.autoFormat_ =
+        this.element.getAttribute('data-auto-format') || '';
+
+    if (this.isResponsive()) {
+      // Attempt to resize to the correct height.
+      const viewport = this.getViewport();
+      return this.attemptChangeSize(
+          AmpAdNetworkAdsenseImpl.getResponsiveHeightForContext_(
+              viewport.getSize()),
+          undefined);
+    }
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    super.onLayoutMeasure();
+
+    if (this.isResponsive() && !this.responsiveAligned_) {
+      this.responsiveAligned_ = true;
+
+      const layoutBox = this.getLayoutBox();
+
+      // Nudge into the correct horizontal position. NB: this must succeed in
+      // order to align the element correctly, so we use changeSize rather than
+      // attemptChangeSize here. But it doesn't actually change the size (it
+      // just changes the horizontal margins).
+      // TODO(charliereams): Is this the right way to get direction?
+      const parentDirection =
+          computedStyle(this.win, this.element)['direction'];
+      const marginChange =
+          parentDirection == 'rtl'
+              ? {right: layoutBox.left}
+              : {left: -1 * layoutBox.left};
+      this.element.getResources()./*OK*/changeSize(
```

Comment:
We just should use `setStyles` for margin change here, combined with `z-index` change. But via `vsync.mutate()`